napari/layers/labels/labels.py

```diff
@@ -708,8 +708,12 @@ def selected_label(self):
     def selected_label(self, selected_label):
         if selected_label == self.selected_label:
             return
-
-        self._prev_selected_label = self.selected_label
+        # when setting the label to the background, store the previous
+        # otherwise, clear it
+        if selected_label == self.colormap.background_value:
+            self._prev_selected_label = self.selected_label
+        else:
+            self._prev_selected_label = None
```

Comment:
will this behave unintuitively if the background label is selected just by (potentially unintentionally) scrolling from `1` to `0` on the spinbox?

Reply:
That's actually one of the main reasons I put this part in, so it works the same whether you used the spinbox or the `swap` method.
Open to other suggestions, i'm mostly working off the end users suggestions and my own usage.